### ISSUE TITLE
Enable Keccak AVX-512 support

### ISSUE THREAD

**Issue by smuellerDD** [openssl/openssl#21126]
After testing, I think that OpenSSL 3.0.8 (and older) as available in current 
Fedora 38, Keccak acceleration with AVX-512 is seemingly not activated. I 
performed a timing with a file that I generated as follows:

dd if=/dev/zero of=/dev/shm/file count=8 bs=1GB

I am maintaining my own crypto library [1]. That library contains an AVX-512 
implementation whose timing is:

$ time build/hash/tests/hasher -h SHA3-512 -f /dev/shm/file
Message digest = 
b2cc3439955e990d319bfc5b0bc1401ae18de07ff275abad5fec77e0f921e8e3d77b9a58b769cd44c7b0fa5c300782a97d151259bd63a76cd16f278c3a4c0126

real    0m54,888s
user    0m53,969s
sys     0m0,762s

Using OpenSSL's implementation on the very same system:

 time openssl dgst -sha3-512 /dev/shm/file
SHA3-512(/dev/shm/file)= 
b2cc3439955e990d319bfc5b0bc1401ae18de07ff275abad5fec77e0f921e8e3d77b9a58b769cd44c7b0fa5c300782a97d151259bd63a76cd16f278c3a4c0126

real    1m20,355s
user    1m18,127s
sys     0m1,984s

After forcing my implementation to use AVX2 (as opposed to AVX-512):

$ time build/hash/tests/hasher -h SHA3-512 -f /dev/shm/file
Message digest = 
b2cc3439955e990d319bfc5b0bc1401ae18de07ff275abad5fec77e0f921e8e3d77b9a58b769cd44c7b0fa5c300782a97d151259bd63a76cd16f278c3a4c0126

real    1m17,026s
user    1m15,906s
sys     0m0,901s

You see, it seems to be the case that AVX-512 support is not active for Keccak.

[1] https://www.chronox.de/leancrypto.html

**Comment by t8m** [comment]
Yes, the AVX2, AVX-512 and other keccak implementations are present in the sources but unused by the build. There would have to be some runtime-selection routine added to utilize them. We do not want to make use of these implementations determined at build time.

**Comment by t8m** [comment]
Given raising pervasiveness of SHA3/SHAKE usage we should really tackle this. @slontis Would you be interested?
